xiaqianZhang/assignments/cyclicSort/lc268/lc268.py

```diff
@@ -0,0 +1,84 @@
+# Problem Statement #
+
+# We are given an array containing ‘n’ distinct numbers taken from the range 0 to ‘n’. 
+# Since the array has only ‘n’ numbers out of the total ‘n+1’ numbers, find the missing number.
+
+#Attempt 1 brainstorm but on the wrong track
+# input: list of distinct integers from range 0-n
+# output: return the missing number
+# given n numbers, and the original length of the array is n+1
+# increment a space in the given array with variable x
+# while loop that swap the numbers if they are not on the right position, currentPtr = swapPtr, index 0 = num 0
+# ignore the x during the swap, and it will gradually end up in some position while the other integers are located at the correct position
+# return the index of where x located, that will be the missing integer
+
+# Solution without using the cyclicSort
+# original Sum = 0-4
+#currentSum = the numbers given from the input array
+#subtract the difference, that will be the missing number
+
+# def find_missing_number(nums):
+#   originalSum = 0
+#   currentSum = 0
+
+#   for each in range(len(nums)+1):
+#     originalSum+=each
+
+#   for each in range(len(nums)):
+#     currentSum+=nums[each]
+  
+#   return originalSum - currentSum
+
+
+#Solution with cyclicSort
+# We are given an array containing ‘n’ distinct numbers taken from the range 0 to ‘n’. 
+# Since the array has only ‘n’ numbers out of the total ‘n+1’ numbers, find the missing number.
+
+def find_missing_number(nums):
+
+
```

Comment:
Missing the coded solution. If you can not come up with the solution on your own, feel free to look at the solution provided and reverse-engineer it.